fdbserver/workloads/ReadHotDetection.actor.cpp

```diff
@@ -0,0 +1,145 @@
+/*
+ * ReadHotDetection.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "fdbrpc/ContinuousSample.h"
+#include "fdbclient/NativeAPI.actor.h"
+#include "fdbserver/TesterInterface.actor.h"
+#include "fdbserver/workloads/BulkSetup.actor.h"
+#include "fdbclient/ReadYourWrites.h"
+#include "fdbserver/workloads/workloads.actor.h"
+#include "flow/actorcompiler.h" // This must be the last #include.
+
+struct ReadHotDetectionWorkload : TestWorkload {
+	int actorCount, rowCount;
+
+	double testDuration, transactionsPerSecond;
+	vector<Future<Void>> clients;
+	Key readKey;
+	KeyRange wholeRange;
+
+	ReadHotDetectionWorkload(WorkloadContext const& wcx) : TestWorkload(wcx) {
+		testDuration = getOption(options, LiteralStringRef("testDuration"), 120.0);
+		transactionsPerSecond = getOption(options, LiteralStringRef("transactionsPerSecond"), 1000.0) / clientCount;
+		actorCount = getOption(options, LiteralStringRef("actorsPerClient"), transactionsPerSecond / 5);
+		rowCount = getOption(options, LiteralStringRef("rowCount"), 1000);
+	}
+
+	virtual std::string description() { return "ReadHotDetection"; }
+
+	virtual Future<Void> setup(Database const& cx) { return _setup(cx, this); }
+
+	virtual Future<Void> start(Database const& cx) {
```

Comment:
I didn't see the workload name is paired with any workload, which means this workload is not triggered in correctness.
Why do we want to exclude this test? 

If it's because of false positive failure, we need to change `keyReader()` to avoid such failures so that we can use  the workload.

Reply:
It's not run because triggering the read hot at the default threshold is kind of hard without changing the knob value. I'd add it back once I figured out an easier way to do that.